### PR TITLE
End to end tests proof of concept

### PR DESCRIPTION
**Pull request by Wonshtrum**
We want to test Sōzu's lib to be sure that all corner cases are checked.
This is intended for local development and inclusion in the CI.

### CODE REVIEW

File: e2e/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+version = "0.1.0"
+edition = "2021"
+
+[[bin]]
+name = "sozu-e2e"
+
+[dependencies]
+sozu-lib = { path = "../lib" }
+sozu-command-lib = { path = "../command" }
+slab = "0.4.7"
```
Comment (FlorentinDUBOIS) [inline e2e/Cargo.toml:12]: Could you add the `^` selector, it does not change the behaviour, but still add more semantic.

File: e2e/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,561 @@
+use http_utils::{http_request, http_response};
```
Comment (FlorentinDUBOIS) [inline e2e/src/main.rs:1]: It is cosmetic, but could you format your input using the following convention:
- one shared `use std::{..};`
- one import per crate
- one import for crate `use crate::{..};`

**Review by FlorentinDUBOIS — CHANGES_REQUESTED** [review]
That's great piece to test Sōzu, but could you provide a command line interface based on `clap` like the command line of Sōzu which allow to select which tests should be run like `cargo tests` command. Besides, I wonder about giving a proper module to each tests. What do you think?

File: e2e/src/main.rs (hunk 2 above)
Comment (Keksoj) [reply e2e/src/main.rs:1]: This!

The order of use statements goes in this order:

1. std
2. extern crates
3. workspaces
4. within-workspace use statements

the `mod` statements should be either above, or underneath, all these.

File: e2e/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+        "{} sent: {}, received: {}",
+        good_client.name, good_client.sent, good_client.received
+    );
+    let aggregator = backend.stop_and_get_aggregator();
+    println!("backend aggregator: {:?}", aggregator);
+}
+
+fn wait_input<S: Into<String>>(s: S) {
+    println!("==================================================================");
+    println!("{}", s.into());
```
Comment (Keksoj) [inline e2e/src/main.rs:486]: this should display some text that says "press any key to continue"

File: e2e/src/sozu/worker.rs [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+            order,
+        });
+    }
+    pub fn read_proxy_response(&mut self) -> Option<ProxyResponse> {
+        self.command_channel.read_message()
+    }
+    pub fn read_to_last(&mut self) {
+        loop {
+            let response = self.read_proxy_response();
+            println!("test received: {:?}", response);
```
Comment (Keksoj) [inline e2e/src/sozu/worker.rs:148]: maybe implement `std::fmt` for response so that the output is more cosmetic than:

```
test received: Some(ProxyResponse { id: "ID_1", status: Ok, content: None })

```

**Review by Keksoj — COMMENTED** [review]
The whole project is awesome, especially the interactive prompt. Go on and good luck!

**Review by Keksoj — APPROVED** [review]
